src/Features/Core/Portable/Completion/CompletionOptions.cs

```diff
@@ -14,6 +14,9 @@ internal static class CompletionOptions
         public static readonly PerLanguageOption<bool> TriggerOnTyping = new PerLanguageOption<bool>(FeatureName, "TriggerOnTyping", defaultValue: true);
         public static readonly PerLanguageOption<bool> TriggerOnTypingLetters = new PerLanguageOption<bool>(FeatureName, "TriggerOnTypingLetters", defaultValue: true);
 
+        public static readonly PerLanguageOption<int> AddNewLineOnEnterAfterFullyTypedWord =
```

Comment:
Good idea on using `EnterKeyRule` instead of `int`.
For the name of the setting, I still need it serialized under the same name.

How about this?
```C#
public static readonly PerLanguageOption<EnterKeyRule> EnterKeyBehavior =
            new PerLanguageOption<EnterKeyRule>(FeatureName, "AddNewLineOnEnterAfterFullyTypedWord", defaultValue: EnterKeyRule.Default);
```